### ISSUE TITLE
Moving your awesome projects under an organization

### ISSUE THREAD

**Issue by wilzbach** [dlang-community/D-Scanner#421]
Hi @Hackerpilot,

I know that you are quite busy, so I am not so sure whether you have seen that here the same (trivial) PR is pending in the queue three times:

![image](https://cloud.githubusercontent.com/assets/4370550/23583877/f14f8022-0150-11e7-8d8a-5f32cb791694.png)

Of course as mentioned before, I am quite aware that your time is limited. However, I was wondering on your thoughts of moving (a) moving your projects under a GH org or (b) giving some key people like @BBasile merge rights, s.t. such simple PRs can be handled without you.

Thanks a lot in advance for considering this.

**Comment by sprinkle131313** [comment]
@Hackerpilot 

I agree with this, simple fixes take far too long to make their way into master. Having more people to maintain the repo will make those one line bug fixes get into master quickly. One bug was fixed, just to find another one and now it's been over a month that has been mostly waiting on pull requests to be merged here before the feature in another project can be implemented.

**Comment by BBasile** [comment]
I agree too. Dscanner is tested in the cloud and in addition the "review required" system could be enabled, which would acts as a barrier against any major error (or too quick self merge). In a first time you could add a contributor to the project and if the situation persists maybe it'll be time for you to think to the organisation.

That being said, @wilzbach and @sprinkle131313 don't forget that because of the submodule system, adding a contributor would not solve all the possible problems, i.e when libdparse is also involved (as shown recently with the _IfCondition_ bug that was not solved in a very straigth way...).

**Comment by wilzbach** [comment]
> That being said, @wilzbach and @sprinkle131313 don't forget that because of the submodule system, adding a contributor would not solve all the possible problems,

That's exactly why I suggested to move all these projects to a guarding GitHub organization:

**Comment by BBasile** [comment]
@wilzbach, can be speak on IRC (d): DotBatch

**Comment by wilzbach** [comment]
> @wilzbach, can we speak on IRC (d): DotBatch

Sorry I am currently on a plane - however feel free to ping me via mail (I will get back to you as soon as I can).

**Comment by BBasile** [comment]
> Sorry I am currently on a plane

I can't complain.

**Comment by wilzbach** [comment]
> Sorry I am currently on a plane - however feel free to ping me via mail (I will get back to you as soon as I can).

@BBasile @sprinkle131313 I am finally back :)
AFAICT atm we are stuck in a limbo and without feedback from @Hackerpilot there aren't many options:
- A: stay with the current "status quo"
- B: fork all Dscanner, libdparse etc. in a maintained GH community organization. Invite @Hackerpilot to the organization or otherwise backport his future work.

What are your thoughts on this? Do you have other ideas?

Btw one related idea is the [creation of a dlang-community GH organisation](https://github.com/d-martians/discussions/issues/2) to avoid the problem of forking a popular library for every small maintenance work. Maybe the same org could be used for Dscanner etc. as well?

**Comment by sprinkle131313** [comment]
Would the name need to be changed for option B? In the case of dub projects and such. It would be preferable if the projects were moved to an organization by @Hackerpilot but it seems he's not even willing to talk about potential solutions for one reason or another. I'd say option B is better than A, I honestly am tired of waiting weeks/months for a single one line patch to get pulled in. I already build and use my own versions of these tools for the most part.

As for the dlang-community organization that would still require @Hackerpilot to give that community merge rights?

**Comment by rikkimax** [comment]
@Hackerpilot has already given all licensing rights required, yay Boost!

The Org will need proper planning, e.g. how do you handle PR's?
Can the author of a PR pull their own ext.

But this should be handled on IRC + a wiki.

**Comment by wilzbach** [comment]
> The Org will need proper planning, e.g. how do you handle PR's?
Can the author of a PR pull their own ext.

> But this should be handled on IRC + a wiki.

Yeah it definitely needs to be documented, but why IRC? 

> Would the name need to be changed for option B? In the case of dub projects and such. It would be preferable if the projects were moved to an organization by @Hackerpilot but it seems he's not even willing to talk about potential solutions for one reason or another.

Yes and yes.

> As for the dlang-community organization that would still require @Hackerpilot to give that community merge rights?

No, forking doesn't require his permissions. However as you mentioned for the dub projects a different name is required.

> B: fork all Dscanner, libdparse etc. in a maintained GH community organization. Invite @Hackerpilot to the organization or otherwise backport his future work.

As we don't seem to make much progress, I did the first step and forked the repos into `dlang-community`, created a team that has write access to these repos and invited everyone who has participated in this discussion to the team.

So in case we don't want to stay with option A (status quo), there are two remaining questions to be solved:
- (from @sprinkle131313) How do we name projects on DUB?
  - Using `-community` as a suffix could work
- (from @rikkimax) How do we handle PRs and merge rights in the future?
  - Naive solution:
    - every non-trivial PR should be approved by someone else
    - contributors who have earned credibility by submitting valuable contributions (can) gain merge rights

**Comment by BBasile** [comment]
I propose to talk on d IRC. Now PLZ.

**Comment by wilzbach** [comment]
@Hackerpilot 

Sorry for being so unresponsive over the last week, unfortunately I can be busy as well :S 
Anyhow, I assigned you full admin rights for the DLang Community organization, so please feel free to delete the forked repositories and move yours to the community :)

![image](https://cloud.githubusercontent.com/assets/4370550/25742939/3246c75e-3192-11e7-93f7-cbb3c38c4206.png)
